app/src/Model/Table/ArticlesTable.php

```diff
@@ -0,0 +1,63 @@
+<?php
+declare(strict_types=1);
+
+// src/Model/Table/ArticlesTable.php
+namespace App\Model\Table;
+
+// the EventInterface class
+use Cake\Event\EventInterface;
+use Cake\ORM\Table;
+// the Text class
+use Cake\Utility\Text;
+// the Validator class
+use Cake\Validation\Validator;
+
+class ArticlesTable extends Table
+{
+    /**
+     * Initialize the table and add timestamp behavior.
+     *
+     * @param array $config Additional table configuration.
+     * @return void
+     */
+    public function initialize(array $config): void
+    {
+        $this->addBehavior('Timestamp');
+    }
+
+    /**
+     * Before saving an entity, generate a slug if it doesn't exist.
+     *
+     * @param \Cake\Event\EventInterface $event Save event.
+     * @param \Cake\Datasource\EntityInterface $entity Entity being saved.
+     * @param array $options Additional options.
+     * @return void
+     */
+    public function beforeSave(EventInterface $event, $entity, $options)
+    {
+        if ($entity->isNew() && !$entity->slug) {
+            $sluggedTitle = Text::slug($entity->title);
+            // trim slug to maximum length defined in schema
+            $entity->slug = substr($sluggedTitle, 0, 191);
+        }
+    }
```

Comment:
If we run the command `docker compose exec php-app composer stan`, we'll get errors within this method as well. The errors as follows:

```
 ------ -------------------------------------------------------------------------------------- 
  Line   Model/Table/ArticlesTable.php                                                         
 ------ -------------------------------------------------------------------------------------- 
  38     Access to an undefined property Cake\Datasource\EntityInterface::$slug.               
         💡 Learn more: https://phpstan.org/blog/solving-phpstan-access-to-undefined-property  
  39     Access to an undefined property Cake\Datasource\EntityInterface::$title.              
         💡 Learn more: https://phpstan.org/blog/solving-phpstan-access-to-undefined-property  
 ------ -------------------------------------------------------------------------------------- 

```

How do we resolve this error?

Reply:
fixed by using method ->get('propertyName')